src/FSharpx.Collections.Experimental/AaTree.fs

```diff
@@ -0,0 +1,184 @@
+namespace rec FSharpx.Collections.Experimental
+
+open System.Collections
+open FSharpx.Collections
+open System.Collections.Generic
+
+(* Implementation guided by following paper: https://arxiv.org/pdf/1412.4882.pdf *)
+
+/// A balanced binary tree similar to a red-black tree which may have less predictable performance.
+type AaTree<'T when 'T: comparison> = 
+    | E
+    | T of int * AaTree<'T> * 'T * AaTree<'T>
```

Comment:
Consider using names for the T case data as in the Shape example here - https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/discriminated-unions .

Reply:
Good idea. Would be helpful when editing the tree itself to specify the different data types mean. (Added.)